### ISSUE TITLE
build failure with Go 1.11 on Amazon Linux

### ISSUE THREAD

**Issue by nmeyerhans** [firecracker-microvm/firecracker-containerd#208]
Agent (at least) fails to build using Go from the golang1.11 "extras" repo on Amazon Linux 2.

```
[ec2-user@ip-10-0-0-119 firecracker-containerd]$ go version
go version go1.11.3 linux/amd64
[ec2-user@ip-10-0-0-119 firecracker-containerd]$ make agent
make -C agent
make[1]: Entering directory `/home/ec2-user/firecracker-containerd/agent'
go build -o agent
go: finding github.com/StackExchange/wmi v0.0.0-20181212234831-e0a55b97c705
go: finding github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2
go: finding github.com/shirou/gopsutil v2.18.12+incompatible
go: finding github.com/containerd/go-runc v0.0.0-20190226155025-7d11b49dc076
...
go: downloading golang.org/x/sys v0.0.0-20190329044733-9eb1bfa1ce65
...
go: downloading golang.org/x/text v0.3.0
go build golang.org/x/sys/unix: module requires Go 1.12
make[1]: *** [agent] Error 1
make[1]: Leaving directory `/home/ec2-user/firecracker-containerd/agent'
make: *** [agent] Error 2
```

`golang.org/x/sys/unix` does build with Go 1.11 on AL2, so at first glance it's not entirely clear what's going on...

**Comment by samuelkarp** [comment]
Looks like that came from this commit: https://github.com/golang/sys/commit/b294cbcfc56d2a65a30c02c78722b5df061e99b8

We can downgrade our version of golang.org/x/sys to before that commit.

**Comment by nmeyerhans** [comment]
I'm inclined to say that this is an AL2 issue, and it can probably be resolved by updating Go. If I use a manually downloaded Go 1.11.11 from golang.org, everything works. (Apparently `go 1.12` in go.mod is satisfied by newer versions of go 1.11.)

For now, I'll try to clarify in our docs that we require the latest patch release of whatever minimum Go version we claim to support.

**Comment by kzys** [comment]
https://golang.org/doc/go1.12#modules

> This changed use of the go directive means that if you use Go 1.12 to build a module, thus recording go 1.12 in the go.mod file, you will get an error when attempting to build the same module with Go 1.11 through Go 1.11.3. Go 1.11.4 or later will work fine, as will releases older than Go 1.11. If you must use Go 1.11 through 1.11.3, you can avoid the problem by setting the language version to 1.11, using the Go 1.12 go tool, via go mod edit -go=1.11. 

Maybe Amazon Linux is having Go <= 1.11.3? Go 1.11.5 (on my Debian box) works as well.

**Comment by kzys** [comment]
Amazon Linux 2's extras now have

```
[ec2-user@ip-172-31-29-52 ~]$ amazon-linux-extras list
  0  ansible2                 available    [ =2.4.2  =2.4.6 ]
  2  httpd_modules            available    [ =1.0 ]
  3  memcached1.5             available    [ =1.5.1  =1.5.16 ]
  4  nginx1.12                available    [ =1.12.2 ]
  5  postgresql9.6            available    [ =9.6.6  =9.6.8 ]
  6  postgresql10             available    [ =10 ]
  8  redis4.0                 available    [ =4.0.5  =4.0.10 ]
  9  R3.4                     available    [ =3.4.3 ]
 10  rust1                    available    \
        [ =1.22.1  =1.26.0  =1.26.1  =1.27.2  =1.31.0 ]
 11  vim                      available    [ =8.0 ]
 13  ruby2.4                  available    [ =2.4.2  =2.4.4 ]
 15  php7.2                   available    \
        [ =7.2.0  =7.2.4  =7.2.5  =7.2.8  =7.2.11  =7.2.13  =7.2.14
          =7.2.16  =7.2.17  =7.2.19 ]
 16  php7.1                   available    \
        [ =7.1.22  =7.1.25  =7.1.27  =7.1.28  =7.1.30 ]
 17  lamp-mariadb10.2-php7.2  available    \
        [ =10.2.10_7.2.0  =10.2.10_7.2.4  =10.2.10_7.2.5
          =10.2.10_7.2.8  =10.2.10_7.2.11  =10.2.10_7.2.13
          =10.2.10_7.2.14  =10.2.10_7.2.16  =10.2.10_7.2.17
          =10.2.10_7.2.19 ]
 18  libreoffice              available    [ =5.0.6.2_15  =5.3.6.1 ]
 19  gimp                     available    [ =2.8.22 ]
 20  docker=latest            enabled      \
        [ =17.12.1  =18.03.1  =18.06.1 ]
 21  mate-desktop1.x          available    [ =1.19.0  =1.20.0 ]
 22  GraphicsMagick1.3        available    [ =1.3.29 ]
 23  tomcat8.5                available    \
        [ =8.5.31  =8.5.32  =8.5.38  =8.5.40  =8.5.42 ]
 24  epel                     available    [ =7.11 ]
 25  testing                  available    [ =1.0 ]
 26  ecs                      available    [ =stable ]
 27  corretto8                available    \
        [ =1.8.0_192  =1.8.0_202  =1.8.0_212  =1.8.0_222 ]
 28  firecracker              available    [ =0.11 ]
 29  golang1.11               available    [ =1.11.3  =1.11.11 ]
 30  squid4                   available    [ =4 ]
 31  php7.3                   available    \
        [ =7.3.2  =7.3.3  =7.3.4  =7.3.6 ]
 32  lustre2.10               available    [ =2.10.5 ]
 33  java-openjdk11           available    [ =11 ]
 34  lynis                    available    [ =stable ]
 35  kernel-ng                available    [ =stable ]
 36  BCC                      available    [ =0.x ]
 37  mono                     available    [ =5.x ]
```

**Comment by nmeyerhans** [comment]
Confirmed that the default "golang1.11" packages on Amazon Linux 2 work.
